alicloud/resource_alicloud_network_interface_test.go

```diff
@@ -0,0 +1,544 @@
+					resource.TestCheckResourceAttrSet("alicloud_network_interface.eni", "vswitch_id"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "security_groups.#", "1"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "private_ip", "192.168.0.2"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "private_ips.#", "5"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "private_ips_count", "5"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "description", "Address count test"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "tags.TF-VER", "0.11.3"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccNetworkInterfaceConfigWithPrivateIpAddressCount2,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudResourceID("alicloud_network_interface.eni"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "name", "tf-testAcc-eni"),
+					resource.TestCheckResourceAttrSet("alicloud_network_interface.eni", "vswitch_id"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "security_groups.#", "1"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "private_ip", "192.168.0.2"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "private_ips.#", "3"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "private_ips_count", "3"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "description", "Address count test2"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "tags.TF-VER", "0.11.3"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAlicloudNetworkInterfaceWithoutPrimaryIpAddress(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+
+		IDRefreshName: "alicloud_network_interface.eni",
+
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckNetworkInterfaceDestroy,
+
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccNetworkInterfaceConfigWithoutPrimaryIpAddress,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudResourceID("alicloud_network_interface.eni"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "name", "tf-testAcc-eni"),
+					resource.TestCheckResourceAttrSet("alicloud_network_interface.eni", "vswitch_id"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "security_groups.#", "1"),
+					resource.TestCheckResourceAttrSet("alicloud_network_interface.eni", "private_ip"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "private_ips.#", "0"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "description", "No primary private IP address test"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "tags.TF-VER", "0.11.3"),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
This is missing the resource exist method

Reply:
Got it